protocols/ledger/ledger.go

```diff
@@ -0,0 +1,141 @@
+package ledger
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+type LedgerManager struct {
+	nonce *big.Int
+}
+
+func NewLedgerManager() LedgerManager {
+	return LedgerManager{
+		nonce: big.NewInt(0),
+	}
+}
+
+// CreateLedger creates a new  two party ledger channel based on the provided left and right outcomes.
+func (l *LedgerManager) CreateLedger(left outcome.Allocation, right outcome.Allocation, secretKey *[]byte, myIndex uint) *channel.TwoPartyLedger {
+
+	leftAddress, _ := left.Destination.ToAddress()
+	rightAddress, _ := right.Destination.ToAddress()
+	initialState := state.State{
+		ChainId:           big.NewInt(9001),
+		Participants:      []types.Address{leftAddress, rightAddress},
+		ChannelNonce:      l.nonce,
+		AppDefinition:     types.Address{},
+		ChallengeDuration: big.NewInt(45),
+		AppData:           []byte{},
+		Outcome: outcome.Exit{outcome.SingleAssetExit{
+			Allocations: outcome.Allocations{left, right},
+		}},
+		TurnNum: 0,
+		IsFinal: false,
+	}
+
+	ledger, lErr := channel.NewTwoPartyLedger(initialState, myIndex)
+	if lErr != nil {
+		panic(lErr)
+	}
+
+	// Update the nonce by 1
+	l.nonce = big.NewInt(0).Add(l.nonce, big.NewInt(1))
+	return ledger
+}
+
+// HandleRequest accepts a ledger request and updates the ledger channel based on the request.
+// It returns a signed state message that can be sent to other participants.
+func (l *LedgerManager) HandleRequest(ledger *channel.TwoPartyLedger, request protocols.LedgerRequest, secretKey *[]byte) (protocols.SideEffects, error) {
+
+	guarantee, _ := outcome.GuaranteeMetadata{
+		Left:  request.Left,
+		Right: request.Right,
+	}.Encode()
+
+	supported, err := ledger.Channel.LatestSupportedState()
+	if err != nil {
+		return protocols.SideEffects{}, fmt.Errorf("Could not find a supported state %w", err)
```

Comment:
```suggestion
		return protocols.SideEffects{}, fmt.Errorf("error finding a supported state: %w", err)
```

error messages are non-capitalized & non-punctuated by convention. Changed the text so that the the `: errorText` reads a little better maybe.